lib/Cake/Test/Case/Model/Datasource/Database/PostgresTest.php

```diff
@@ -327,7 +327,7 @@ public function testColumnParsing() {
  * @return void
  */
 	public function testValueQuoting() {
-		$this->assertEquals("1.200000", $this->Dbo->value(1.2, 'float'));
+		$this->assertEquals("1.2", $this->Dbo->value(1.2, 'float'));
```

Comment:
This change addresses the following test failures.

```
There were 2 failures:

1) PostgresTest::testValueQuoting
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'1.200000'
+'1.2'

/home/runner/work/cakephp2-php8/cakephp2-php8/lib/Cake/Test/Case/Model/Datasource/Database/PostgresTest.php:330

2) PostgresTest::testLocalizedFloats
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'3.140000'
+'3.14'

/home/runner/work/cakephp2-php8/cakephp2-php8/lib/Cake/Test/Case/Model/Datasource/Database/PostgresTest.php:372
```

I examined the commit history to determine if the above test failure was indicative of a faulty implementation or a problem with the test code.

It seems that there was a time after CakePHP 1.3.4 when the implementation of `DboPostgres::value(float value)` was formatting with `sprintf('%F', $data)`. 
[committed on Aug 21, 2010 -- "Fixing failing tests in DboMysql and DboPostgres related to localized floats."](https://github.com/cakephp/cakephp/commit/12d4b52f105e6c84824eb5c8e6ca6d4996760e4d#diff-a6ae8632304977724bfeff550d23603ad1ce3c00a04d8d3a021f55d734d06da8R302-R305)
[committed on Aug 21, 2010 -- "Adding tests to dbo_postgres for localized floats."](https://github.com/cakephp/cakephp/commit/eb96d8acfbe6c31230fb28f21110aaec5de57eaf)
```php
sprintf('%F', 1.2) // => '1.200000'
```
(Note: The `assertIdentical()` in the test code is an assertion method of SimpleTest, a unit testing framework used in CakePHP 1.x. It is the same as `assertSame()` in PHPUnit.)

However, during the development of CakePHP 2.0, when the implementation was merged into `DboSource::value()`, the assertion method was changed to `assertEqual()`. (This is equivalent to `assertEquals()` in PHPUnit.)
[committed on Oct 18, 2010 -- "Refactoring DboSource::value() so postgres and mysq behavior is the same, updating test case"](https://github.com/cakephp/cakephp/commit/d9c9a32ff3d2994d7c4a0557b9be15f082cbc4f1#diff-05225941bb88bdd2f1ff4ca87f37d1b725949ca290e12899e1e93c8739f531f3R333)

After that, the implementation of `DboSource::value(float value)` was changed to `str_replace(','', '. , sprintf('%G', $data))`.
However, the expected value of the assertion was not changed at this time.
[committed on Sep 19, 2011 -- "Merge branch '1.3' into merger"](https://github.com/cakephp/cakephp/commit/6a4e7558fcceb06ec05f345ba85c0ea54f36c39e#diff-892446cee249a140b8b571457211eb1b6dab8077c928b68d80727f25c8a53315R331)
```php
str_replace(',', '.', sprintf('%G', 1.2)) // => '1.2'
```

And later, improvements to `sebastian/comparator`, on which PHPUnit depends, caused the test to fail.
Finally, in `sebastian/comparator:">=3.0.1"`, if `$expected` and `$actual` are both strings, they are now compared using the `!==` instead of the `!=`, which is why the test failed.
[sebastianbergmann/comparator ChangeLog.md 3.0.1 - 2018-06-14](https://github.com/sebastianbergmann/comparator/blob/72f01e6586e0caf6af81297897bd112eb7e9627c/ChangeLog.md#301---2018-06-14)